subprojects/signer/src/main/kotlin/com/avito/plugin/SignServicePlugin.kt

```diff
@@ -21,71 +21,89 @@ public class SignServicePlugin : Plugin<Project> {
 
         val extension = target.extensions.create<SignExtension>("signService")
 
+        // AGP 4.2 variant
+        // target.extensions.getByType<ApplicationAndroidComponentsExtension>().run {
+
         target.withAndroidApp { appExtension ->
 
+            // AGP 4.2 remove this block completely
+            appExtension.applicationVariants.all { variant: com.android.build.gradle.api.ApplicationVariant ->
+
+                val buildTypeName = variant.buildType.name
+                val apkToken: String? = extension.apkSignTokens[buildTypeName]
+                val bundleToken: String? = extension.bundleSignTokens[buildTypeName]
+
+                variant.outputsAreSigned = apkToken.hasContent() || bundleToken.hasContent()
+
+                target.tasks.signedApkTaskProvider(variant.name).configure { signApkTask ->
+                    signApkTask.dependsOn(variant.packageApplicationProvider)
```

Comment:
you should not have to do this. Since you are using the new variant api to wire the task input to the APK, it will be automatically a dependent.

Reply:
Yes, you're right, removed this manual wiring;

What about `variant.outputsAreSigned`? 

It's not deprecated, but look like it should be somehow reflected in new API's, right?

How to say properly to AGP(post 7.0) that i want to handle signing myself?